src/reconcile/rec_visibility.c

```diff
@@ -437,7 +458,7 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
      *
      * Additionally history store reconciliation is not set skip saving an update.
      */
-    if (__rec_need_save_upd(session, r, upd_select, has_newer_updates)) {
+    if (upd_select->prepare || __rec_need_save_upd(session, r, upd_select, has_newer_updates)) {
```

Comment:
Can we merge this logic in __rec_need_save_upd?

Reply:
Fixed.